### PR TITLE
feat(sdk): Allow changing a device's display name

### PR DESCRIPTION
**Pull request by dasJ**
Allows me to rename a device.

Signed-off-by:

### CODE REVIEW

File: crates/matrix-sdk/src/client/mod.rs [hunk 1]
```diff
@@ -1928,6 +1928,26 @@ impl Client {
...
     }
 
+    /// Change the display name of a device owned by the current user.
+    ///
+    /// Returns a `update_device::Response` which specifies the result
+    /// of the operation.
+    ///
+    /// # Arguments
+    ///
+    /// * `device` - The `Device` to change the display name of.
```
Comment (poljar) [inline crates/matrix-sdk/src/client/mod.rs:1938]: `device` should rather be `device_id`, in the rust-sdk a `Device` is a higher level object, not just the identifier: https://matrix-org.github.io/matrix-rust-sdk/matrix_sdk/encryption/identities/struct.Device.html

We're giving the ID of the device we're changing the display name of, not the `Device.

File: crates/matrix-sdk/src/client/mod.rs [hunk 2]
```diff
@@ -1928,6 +1928,26 @@ impl Client {
...
+    /// * `display_name` - The new display name to set.
+    pub async fn rename_device(
+        &self,
+        device: &OwnedDeviceId,
```
Comment (poljar) [inline crates/matrix-sdk/src/client/mod.rs:1942]: Same as above, let's call the parameter `device_id`.

**Review by poljar — CHANGES_REQUESTED** [review]
Looks good, apart from the `Device`/`DeviceId` confusion.

**Comment by dasJ** [comment]
Sorry, I forgot rustfmt after amending the change :/

File: crates/matrix-sdk/src/client/mod.rs (hunk 2 above)
Comment (jplatte) [reply crates/matrix-sdk/src/client/mod.rs]: Also, this should be `&DeviceId`. `OwnedDeviceId` derefs to `DeviceId`.

**Review by poljar — APPROVED** [review]
Thanks, looks good.

**Pull request merged**